### DISCUSSION TITLE
Can't get hit object on reality data in itwin

### DISCUSSION THREAD

**Discussion by tusharsable98** [iTwin/itwinjs-core#5533] [General]
Hi All,

I am using the following code to get hit points on onclick event of a tool.

const hit = await IModelApp.locateManager.doLocate(
      new LocateResponse(),
      true,
      ev.point,
      ev.viewport,
      ev.inputSource
    );

but I can't get the hit object when clicked on reality data.Is there any way to get the hit point on reality data

**Comment by bbastings** [comment]
Context reality models that are part of a display style get assigned a transient id and are treated like view decorations by locate. In order for your tool to locate transient geometry you will need to set:

IModelApp.locateManager.options.allowDecorations = true;

Establishing a tool's locate options is typically done in it's onPostInstall method. You can then use HitDetail.isModelHit to check if the hit represents reality data.
